### PR TITLE
#166214086 heroku scripts

### PR DESCRIPTION
**Pull request by Nennyfills**
What does this PR do?
Deploy an application to Heroku

Description of Task to be completed?
- Add build script to the package.json file
- Modify start script to the package.json file
- Add engines version to the package.json file 
- Add Heroku-post-build script
- Create Heroku Procfile file

How should this be manually tested?
N/A

Any background context you want to provide?
N/A

Heroku Link
[hello-books-be-staging](hello-books-be-staging.herokuapp.com)
[hello-books-be](hello-books-be.herokuapp.com)

What are the relevant pivotal tracker stories?
[#166214086](https://www.pivotaltracker.com/story/show/166214086)

Screenshots (if appropriate)
N/A

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: npm start
```
Comment (allebd) [inline Procfile:1]: A newline is required at the end of the file

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
     "description": "A system that allows efficient, cost-effective management of our library",
     "main": "index.js",
     "scripts": {
-        "start": "node server/index.js",
+        "build": "babel ./server -d build",
+        "start": "babel  ./server -d build && node build/index.js",
```
Comment (allebd) [inline package.json]: You can use npm run build on the start script instead of the duplicate statements

File: package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
     "description": "A system that allows efficient, cost-effective management of our library",
     "main": "index.js",
     "scripts": {
-        "start": "node server/index.js",
+        "build": "babel ./server -d build",
```
Comment (KvNGCzA) [inline package.json]: build into a folder called `dist`.

File: package.json [hunk 4]
```diff
@@ -4,7 +4,8 @@
...
+        "start": "babel  ./server -d build && node build/index.js",
         "start:dev": "nodemon --exec babel-node server/index.js",
```
Comment (KvNGCzA) [inline package.json:9]: Add a heroku-postbuild script also

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

**Review by Lundii — COMMENTED** [review]
You have merge conflicts in your package.json file. Please resolve conflicts.

File: Procfile (hunk 1 above)
Comment (Nennyfills) [reply Procfile:1]: done

File: package.json (hunk 2 above)
Comment (Nennyfills) [reply package.json]: The correction has been made please review this again.

File: package.json (hunk 3 above)
Comment (Nennyfills) [reply package.json]: I have corrected that, please review this again.

**Review by Lundii — APPROVED** [review]

**Pull request closed without merging**